onnxruntime/core/graph/graph_utils.h

```diff
@@ -50,12 +50,10 @@ cannot be overridden at runtime. If the initializer is not found or is not const
 const ONNX_NAMESPACE::TensorProto* GetConstantInitializer(const Graph& graph, const std::string& name,
                                                           bool check_outer_scope = true);
 
-/** Find the initializer called 'original_name' in 'graph', or its ancestors if check_outer_scope is true, 
-    and replace with 'initializer' in the current graph. 
-    Does NOT look in any subgraphs. Requires original_name to match an initializer.
-    */
-void ReplaceInitializer(Graph& graph, const std::string& original_name, const ONNX_NAMESPACE::TensorProto& initializer,
-                        bool check_outer_scope = true);
+/** Add a new constant initializer to 'graph'. Checks that new_initializer does not already exist in 'graph'. 
+@returns The NodeArg for the new initializer. 
+*/
+NodeArg& AddConstantInitializer(Graph& graph, const ONNX_NAMESPACE::TensorProto& new_initializer);
```

Comment:
If it was added to the graph inputs that would make it mutable. The 'constant' comes from it being generally used with IsConstantInitializer and GetConstantInitializer. It could however just be called AddInitializer.

Reply:
I'll rename and clarify the comments.

Reply:
Yes my bad about the graph inputs.
I think AddInitializer is better as a name. The ConstantInitializer makes me thing of all the discussion of whether it is in the graph inputs, IR versions :) and so on, and I think it does not add something here.